### PR TITLE
Add `lc90` and `adj123` fields to ancillary netcdf file

### PR DESCRIPTION
**Pull request by trey-stafford**
These are required by the NT2 land spillover.

Some possible TODOs:

* Rename variables to be more descriptive
* Add other attributes e.g., `standard_name` as appropriate?

### CODE REVIEW

File: scripts/gather_masks.py [hunk 1]
```diff
@@ -214,8 +293,54 @@ def ecdr_min_ice_concentration_12km(
...
+
+        l90c_da = xr.DataArray(
+            data=l90c.copy(),
+            coords=dict(
+                y=surfgeo_ds.y.data,
+                x=surfgeo_ds.x.data,
+            ),
+            attrs=dict(
+                grid_mapping="crs",
+                comment=(
```
Comment (trey-stafford) [inline scripts/gather_masks.py:316]: Added the docstring from `create_land90_conc_file` as a `comment`. 

Are there other attrs we should provide here, to make it clearer what this field is and what its used for?

File: scripts/gather_masks.py [hunk 2]
```diff
@@ -214,8 +293,54 @@ def ecdr_min_ice_concentration_12km(
...
+
+        adj123_da = xr.DataArray(
+            data=adj123.copy(),
+            coords=dict(
+                y=surfgeo_ds.y.data,
+                x=surfgeo_ds.x.data,
+            ),
+            attrs=dict(
+                grid_mapping="crs",
+                comment="Diagonal adjacency 123 field",
```
Comment (trey-stafford) [inline scripts/gather_masks.py:336]: TODO: better comment for the `adj123` variable. Other attrs that would make it more clear what this field is?

File: scripts/gather_masks.py [hunk 3]
```diff
@@ -214,8 +293,54 @@ def ecdr_min_ice_concentration_12km(
...
+            ),
+        )
+
         data_vars = dict(
             crs=surfgeo_ds.crs,
+            l90c=l90c_da,
+            adj123=adj123_da,
```
Comment (trey-stafford) [inline scripts/gather_masks.py:343]: TODO: better variable names?

**Review by sc0tts — APPROVED** [review]
This looks good to me.

**Comment by trey-stafford** [comment]
I created a separate task for reviewing these variables' attributes/names: https://trello.com/c/XLKgF7hi/331-ecdr-ancillary-fields-lc90-and-adj123-review-metadata

**Pull request merged**